plutus-core/plutus-ir/src/PlutusIR/Transform/CommuteFnWithConst.hs

```diff
@@ -0,0 +1,130 @@
+        _                   -> tm
+    localCommute tm = tm
+
+commuteFnWithConst :: forall tyname name uni fun a. Typeable fun =>
+    Term tyname name uni fun a -> Term tyname name uni fun a
+commuteFnWithConst = case eqT @fun @DefaultFun of
+    Just Refl -> commuteDefaultFun
+    Nothing   -> id
+
+-- | Returns whether a `DefaultFun` is commutative with `Constant`'s as arguments. Not using
+-- catchall to make sure that this function catches newly added `DefaultFun`.
+isCommutativeWithConstant :: DefaultFun -> Bool
+isCommutativeWithConstant = \case
+  AddInteger                      -> False
+  SubtractInteger                 -> False
+  MultiplyInteger                 -> True
+  DivideInteger                   -> False
+  QuotientInteger                 -> False
+  RemainderInteger                -> False
+  ModInteger                      -> False
+  EqualsInteger                   -> True
+  LessThanInteger                 -> False
+  LessThanEqualsInteger           -> False
+    -- Bytestrings
+  AppendByteString                -> False
+  ConsByteString                  -> False
+  SliceByteString                 -> False
+  LengthOfByteString              -> False
+  IndexByteString                 -> False
+  EqualsByteString                -> True
+  LessThanByteString              -> False
+  LessThanEqualsByteString        -> False
+    -- Cryptography and hashes
+  Sha2_256                        -> False
+  Sha3_256                        -> False
+  Blake2b_256                     -> False
+  VerifyEd25519Signature          -> False
+  VerifyEcdsaSecp256k1Signature   -> False
+  VerifySchnorrSecp256k1Signature -> False
+    -- Strings
+  AppendString                    -> False
+  EqualsString                    -> True
+  EncodeUtf8                      -> False
+  DecodeUtf8                      -> False
+    -- Bool
+  IfThenElse                      -> False
+    -- Unit
+  ChooseUnit                      -> False
+    -- Tracing
+  Trace                           -> False
+    -- Pairs
+  FstPair                         -> False
+  SndPair                         -> False
+    -- Lists
+  ChooseList                      -> False
+  MkCons                          -> False
+  HeadList                        -> False
+  TailList                        -> False
+  NullList                        -> False
+    -- Data
+  ChooseData                      -> False
+  ConstrData                      -> False
+  MapData                         -> False
+  ListData                        -> False
+  IData                           -> False
+  BData                           -> False
+  UnConstrData                    -> False
+  UnMapData                       -> False
+  UnListData                      -> False
+  UnIData                         -> False
+  UnBData                         -> False
+  EqualsData                      -> False -- doesn't take constant
```

Comment:
I don't understand what you're saying here. It does take constants. And it is commutative.